backends/fluidreview.py

```diff
@@ -0,0 +1,136 @@
+"""
+FluidReview API backend
+"""
+from urllib.parse import urljoin
+from datetime import timedelta
+from django.conf import settings
+from requests_oauthlib import OAuth2Session
+
+from backends.models import OAuthToken
+from backends.utils import utc_now
+
+API_NAME = 'FluidReview'
+API_BASE_URL = 'https://bootcampmit.fluidreview.com/api/v2/'
+API_REFRESH_URL = urljoin(API_BASE_URL, 'o/token/')
+
+
+class FluidReviewAPI:
+    """
+    Class for making authorized requests to the FluidReview API via OAuth2
+    """
+
+    def __init__(self):
+        """
+        Prepare to make OAuth2 requests to the API
+        """
+        self.session = self.initialize_session()
+
+    def initialize_session(self):
+        """
+        Initialize an OAuth2 session with an auto-refreshing token.
+        """
+        token = self.get_token()
+        extra = {
+            'client_id': settings.FLUIDREVIEW_CLIENT_ID,
+            'client_secret': settings.FLUIDREVIEW_CLIENT_SECRET,
+        }
+        return OAuth2Session(settings.FLUIDREVIEW_CLIENT_ID,
+                             token=token,
+                             auto_refresh_kwargs=extra,
+                             auto_refresh_url=API_REFRESH_URL,
+                             token_updater=self.save_token)
+
+    def get_token(self):
+        """
+        Return an OAuth token for the API. Search for an OAuthToken model first,
+        and if not found then seed from the initial settings.
+        """
+        token = OAuthToken.objects.filter(api=API_NAME).first()
+        if not token:
+            token = OAuthToken.objects.create(
+                api=API_NAME,
+                access_token=settings.FLUIDREVIEW_ACCESS_TOKEN,
+                refresh_token=settings.FLUIDREVIEW_REFRESH_TOKEN,
+                token_type='Bearer',
+                expires_on=utc_now()
+            )
+        return token.json
+
+    def save_token(self, new_token):
+        """
+        Create or update the FluidReview token parameters.
+        Should be automatically called when a new token is required.
+        With the FluidReview API, the refresh token can only be used once and is then invalidated,
+        so it must be saved and updated in the database.
+
+        Args:
+            new_token(dict): New token returned by the FluidReview API
+
+        Returns:
+            (OAuthToken) the saved object
```

Comment:
The syntax for return values should be: `OAuthToken: the saved object`. Can you change it here and elsewhere in this file?

Reply:
OK